gis/text_to_centreline/sql/helper_functions/function-translate_intersection_point.sql

```diff
@@ -4,17 +4,27 @@ CREATE OR REPLACE FUNCTION gis._translate_intersection_point(
 )
 RETURNS geometry AS $translated_geom$
 DECLARE
+direction text := trim(lower(_translate_intersection_point.direction));
 translated_geom text := (
-    (CASE WHEN TRIM(direction) = 'west' THEN ST_Translate(oid_geom, -metres, 0)
-    WHEN TRIM(direction) = 'east' THEN ST_Translate(oid_geom, metres, 0)
-    WHEN TRIM(direction) = 'north' THEN ST_Translate(oid_geom, 0, metres)
-    WHEN TRIM(direction) = 'south' THEN ST_Translate(oid_geom, 0, -metres)
-    WHEN TRIM(direction) IN ('southwest', 'south west') THEN ST_Translate(oid_geom, -cos(45)*metres, -sin(45)*metres)
-    WHEN TRIM(direction) IN ('southeast', 'south east') THEN ST_Translate(oid_geom, cos(45)*metres, -sin(45)*metres)
-    WHEN TRIM(direction) IN ('northwest', 'north west') THEN ST_Translate(oid_geom, -cos(45)*metres, sin(45)*metres)
-    WHEN TRIM(direction) IN ('northeast', 'north east') THEN ST_Translate(oid_geom, cos(45)*metres, sin(45)*metres)
-    END)
-    );
+    CASE
+        WHEN regexp_instr(direction, '(?<!\S)(w|west)(?!\S)') = 1
+        THEN ST_Translate(oid_geom, -metres, 0)
+        WHEN regexp_instr(direction, '(?<!\S)(e|east)(?!\S)') = 1
+        THEN ST_Translate(oid_geom, metres, 0)
+        WHEN regexp_instr(direction, '(?<!\S)(n|north)(?!\S)') = 1
+        THEN ST_Translate(oid_geom, 0, metres)
+        WHEN regexp_instr(direction, '(?<!\S)(s|south)(?!\S)') = 1
+        THEN ST_Translate(oid_geom, 0, -metres)
+        WHEN direction = 'southwest'
```

Comment:
is it ok to get rid of possibility of matching to ones with space in between words? e.g south west

Reply:
the translated point is different, also some weird results from the one with space between north and east :meow_dio:
```
SELECT tests.*, clean.*
FROM (
    VALUES
		(4, 'BRIMORTON DRIVE', 'A POINT 101 METRES south WEST OF AMBERJACK BOULEVARD AND A POINT 103 METRES north East OF DOLLY VARDEN BOULEVARD', NULL),
        (5, 'BRIMORTON DRIVE', 'A POINT 101 M southwest OF AMBERJACK BOULEVARD AND A POINT 103 M northeast OF DOLLY VARDEN BOULEVARD', NULL)
        --(1, 'Richland Crescent', 'Entire Length', NULL),
        --(2, 'Foxridge Drive', 'Birchmount Road and Kennedy Road', NULL),
        --(3, 'Ravenwood Place', 'Ferris Road and a point 64.11 metres southwest', NULL),
        --(4, 'Brimorton Drive', 'A point 101 metres soutwest of Amberjack Boulevard and a point 103 metres east of Dolly Varden Boulevard', NULL)
    ) AS tests (_bylaw_id, highway, frm, "to"),
LATERAL (
    SELECT *
    FROM gwolofs.text_to_centreline(tests._bylaw_id, tests.highway, tests.frm, tests."to")
) AS clean
ORDER BY 2, 1
```

Reply:
I adjusted the `custom_case` function to clean up all the directions with spaces in them!

Now these both work: 🥳 
```sql
WITH val (_bylaw_id, highway, frm, "to") AS (
    (VALUES 
        (4, 'BRIMORTON DRIVE', 'A POINT 101 METRES south WEST OF AMBERJACK BOULEVARD AND A POINT 103 METRES north East OF DOLLY VARDEN BOULEVARD', NULL),
        (5, 'BRIMORTON DRIVE', 'A POINT 101 M southwest OF AMBERJACK BOULEVARD AND A POINT 103 M northeast OF DOLLY VARDEN BOULEVARD', NULL)
    )
)

SELECT 'gwolofs', * FROM val, gwolofs.text_to_centreline(val._bylaw_id, val.highway, val.frm, val."to")
```